src/app/wallets/update-pending-payments.ts

```diff
@@ -0,0 +1,187 @@
+import { InconsistentDataError } from "@domain/errors"
+import { toLiabilitiesAccountId } from "@domain/ledger"
+import { FeeReimbursement } from "@domain/ledger/fee-reimbursement"
+import { LedgerService } from "@services/ledger"
+import { LndService } from "@services/lnd"
+import { LockService } from "@services/lock"
+import { PriceService } from "@services/price"
+
+export const updatePendingPayments = async ({
+  walletId,
+  logger,
+  lock,
+}: {
+  walletId: WalletId
+  logger: Logger
+  lock?: DistributedLock
+}): Promise<void | ApplicationError> => {
+  const liabilitiesAccountId = toLiabilitiesAccountId(walletId)
+
+  const lockService = LockService()
+  await lockService.lockWalletId({ walletId, logger, lock }, async () => {
+    const ledgerService = LedgerService()
+    const pendingPaymentTransactions = await ledgerService.listPendingPayments(
+      liabilitiesAccountId,
+    )
+    if (pendingPaymentTransactions instanceof Error) return pendingPaymentTransactions
+
+    for (const paymentLiabilityTx of pendingPaymentTransactions) {
+      await updatePendingPayment({ liabilitiesAccountId, paymentLiabilityTx, logger })
+    }
+  })
+}
+
+const updatePendingPayment = async ({
+  liabilitiesAccountId,
+  paymentLiabilityTx,
+  logger,
+}: {
+  liabilitiesAccountId: LiabilitiesAccountId
+  paymentLiabilityTx: LedgerTransaction
+  logger: Logger
+}): Promise<void | ApplicationError> => {
+  const paymentLogger = logger.child({
+    topic: "payment",
+    protocol: "lightning",
+    transactionType: "payment",
+    onUs: false,
+    payment: paymentLiabilityTx,
+  })
+
+  const lndService = LndService()
+  if (lndService instanceof Error) return lndService
+
+  const { paymentHash, pubkey } = paymentLiabilityTx
+  // If we had PaymentLedgerType => no need for checking the fields
+  if (!paymentHash)
+    return new InconsistentDataError("paymentHash missing from payment transaction")
+  if (!pubkey) return new InconsistentDataError("pubkey missing from payment transaction")
+
+  const lnPaymentLookup = await lndService.lookupPayment({
+    pubkey,
+    paymentHash,
+  })
+  if (lnPaymentLookup instanceof Error) {
+    const lightningLogger = logger.child({
+      topic: "payment",
+      protocol: "lightning",
+      transactionType: "payment",
+      onUs: false,
+    })
+    lightningLogger.error({ err: lnPaymentLookup }, "issue fetching payment")
+    return lnPaymentLookup
+  }
+  const { status, roundedUpFee } = lnPaymentLookup
+
+  if (status === PaymentStatus.Settled || status === PaymentStatus.Failed) {
+    const ledgerService = LedgerService()
+    const settled = await ledgerService.settlePendingLiabilityTransactions(paymentHash)
+    if (settled instanceof Error) {
+      paymentLogger.error({ error: settled }, "we didn't have any transaction to update")
+      return settled
+    }
+
+    // PaymentStatus.Failed transactions have null 'roundedUpFee'
+    if (status === PaymentStatus.Settled && roundedUpFee !== null) {
```

Comment:
same here, why not just set 0 (instead of null) and validate here `roundedUpFee > 0`

Reply:
a `roundedUpFee` of `0` instead of `undefined` might trigger a false `feeReimbursement`, or?